lib/chef/azure/service.rb

```diff
@@ -155,4 +96,140 @@ def is_running?
     end
     return false
   end
+
+  def add_or_update_interval_in_client_rb(client_rb, new_chef_service_interval)

```

Comment:
I'd move this out of `service` and into enable.rb or its own class. Editing client.rb configuration really is not a concern of the service.